frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -142,9 +142,13 @@ export default class FirebaseInteractor {
     let idToUse = id || this.auth.currentUser?.uid;
     let user = await this.db.collection("users").doc(idToUse).get();
     let userData = user.data();
-    if (idToUse != null && userData != null) {
-      return this.getUserFromData(idToUse, userData);
+    if (userData != undefined) {
```

Comment:
got the ts error saying that this should be !==

Reply:
got these compilation errors actually:

![image](https://github.com/sandboxnu/vocab-buddy/assets/72981805/c2d5ce13-3132-4bf9-b23c-91d9401ee811)